### PR TITLE
Feature/5780 search request policy

### LINKED ISSUE #5780: Update PoolCandidateSearchRequestPolicy, PoolCandidateFilterPolicy, ApplicantFilterPolicy to use new permissions
## ✨ Feature

A sub-task of #5438.

Our Laravel Policies should check for our new permissions to decide if a user can do something, instead of checking `$user->isAdmin`. 

## 🕵️ Details

[Permissions table](https://docs.google.com/spreadsheets/d/1RC7wufxwjLEvaEunHsfZfOzBefg218dvvzkDwKSiQjI/edit?usp=sharing)

This sub-task cares about how PoolCandidateSearchRequests are protected. 

Note: PoolCandidateFilters and ApplicantFilters are always part of a PoolCandidateSearchRequest, and therefore their policy methods can query the related policy method for their PoolCandidateSearchRequest. In the long run, I suspect we can remove any top-level queries for the Filter types, and remove their Policies entirely.

Expect to work with the following permissions:
- create-any-searchRequest
- view-any-searchRequest
- update-any-searchRequest
- delete-any-searchRequest

## 🙋‍♀️ Proposed Implementation

For some existing queries or mutations, particularly some using a viewAny directive, it may be impossible to re-implement the existing policy method using our [new permissions](https://docs.google.com/spreadsheets/d/1RC7wufxwjLEvaEunHsfZfOzBefg218dvvzkDwKSiQjI/edit?usp=sharing). For example, `pools: [Pool]! @all @can(ability: "viewAny")`. Our new permissions table doesn't include a view-any-pool permission. In this case, leave a `// TODO` comment in the Policy method and deprecate the Api query/mutation, with a comment explaining this is no longer allowed under the new Teams-based roles.

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

- [ ]  PoolCandidate policy methods now work based on the new permissions, instead of the `isAdmin` role
- [ ] Roles follow the permissions in the excel sheet https://docs.google.com/spreadsheets/d/1RC7wufxwjLEvaEunHsfZfOzBefg218dvvzkDwKSiQjI/edit#gid=428263758 
- [ ] All top-level mutations and queries for PoolCandidates have a `@can` directive
- [ ] For any queries whose policy methods can't be implement with new permissions, deprecate the query/mutation and add a `TODO: delete me when my query/mutation gets removed` in the required policy method.
- [ ] php unit tests to verify the policy methods by user roles

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #5780 

## 👋 Introduction

Update PoolCandidateSearchRequest policy

## 🕵️ Details

- tweak policy file to use ableTo()
- nest applicantFilter and poolCandidateFilter policy calls in the aforementioned file
- powered up `admin@test.com`, we're used to that one as the do-anything one
- add a bit of testing 

## 🧪 Testing

1. ensure phpunit succeeds
2. can submit anonymous talent requests
3. can view requests as admin

### CODE REVIEW

**Review by tristan-orourke — CHANGES_REQUESTED** [review]

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -1656,6 +1658,11 @@ type Mutation {
     @update
     @guard
     @can(ability: "update", find: "id")
+  deletePoolCandidateSearchRequest(id: ID!): PoolCandidateSearchRequest
+    @delete(globalId: false)
+    @guard
+    @can(ability: "delete", find: "id")
+
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql:1665]: we were adding a permission for a purpose of request deletion, so I thought it was needed

image

File: api/graphql/schema.graphql [hunk 2]
```diff
@@ -1646,6 +1646,8 @@ type Mutation {
     @delete(globalId: false)
     @guard
     @can(ability: "delete", find: "id")
+
+  # search requests
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql:1650]: we've already started section-ing the mutations list, so I thought to continue along with it

https://github.com/GCTC-NTGC/gc-digital-talent/blob/f3d68dd6dcfa40e6fa8c04f355c908a9a9e4001f/api/graphql/schema.graphql#L1736

https://github.com/GCTC-NTGC/gc-digital-talent/blob/f3d68dd6dcfa40e6fa8c04f355c908a9a9e4001f/api/graphql/schema.graphql#L1774-L1786

File: api/graphql/schema.graphql [hunk 3]
```diff
@@ -956,20 +956,20 @@ type Query {
   poolCandidateFilter(id: UUID! @eq): PoolCandidateFilter
     @find
     @guard
-    @can(ability: "view", query: true)
+    @can(ability: "view", model: "PoolCandidateSearchRequest", query: true)
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql]: done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/5814/commits/f98c3bc5dbcdc5c805380487a0a737b5556d8a9c

File: api/graphql/schema.graphql (hunk 1 above)
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply api/graphql/schema.graphql:1677]: Ah, good point.

File: api/graphql/schema.graphql (hunk 2 above)
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply api/graphql/schema.graphql:1662]: Fair. In that case I recommend following the format: `# SearchRequest mutations`

**Review by tristan-orourke — APPROVED** [review]

File: api/graphql/schema.graphql (hunk 2 above)
[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql]: aye aye
https://github.com/GCTC-NTGC/gc-digital-talent/pull/5814/commits/d341fbd6f599201b6e01799a87a0de612a07aef4

**Pull request merged**